src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,166 @@
+ */
+public class Deadline extends Task {
+    private LocalDate by;
+
+    /**
+     * constructor
+     * @param descrip the dicription of the task.
+     * @param by by field.
+     */
+    public Deadline(String descrip, LocalDate by) {
+        super(descrip);
+        this.by = by;
+    }
+
+    public Deadline(String descrip, ArrayList<String> tags, LocalDate by) {
+        super(descrip, tags);
+        this.by = by;
+    }
+
+    /**
+     * Override the abstract class
+     * @return T
+     */
+    @Override
+    public String getTaskTypeIcon() {
+        return "D";
+    }
+
+    /**
+     * Override the abstract class
+     * @return deadline
+     */
+    @Override
+    public String getTaskType() {
+        return "deadline";
+    }
+
+    /**
+     * Override the toString method
+     * @return String representation of the task
+     */
+    @Override
+    public String toString() {
+        return String.format("%s(by: %s)", super.toString(),
+                this.by.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
+    }
+
+    /**
+     * return by
+     * @return return by
+     */
+    public String getBy() {
+        return this.by.toString();
+    }
+
+    /**
+     * Whether we have to start doing it
+     * @param current current time
+     * @return yes/no
+     */
+    public boolean isTimeForStart(LocalDate current) {
+        if (isDone) {
+            return false;
+        }
+        return current.compareTo(by) <= 0;
+    }
+
+    public LocalDate getByTime() {
+        return this.by;
+    }
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof Deadline) {
+            Deadline temp = (Deadline) obj;
+            boolean equalDescribe = this.description.equals(temp.getDescription());
+            boolean equalBy = this.by.isEqual(temp.by);
+            return equalDescribe && equalBy;
+        }
+        return false;
+    }
+
+    @Override
+    public int compareTo(Task otherTask) {
+        if (otherTask instanceof Todo) {
+            return -1;
+        }
+        if (otherTask instanceof Event) {
+            Event temp = (Event) otherTask;
+            return this.by.compareTo(temp.getFromTime());
+        }
+        Deadline temp = (Deadline) otherTask;
+        return this.by.compareTo(temp.getByTime());
+    }
+
+    @Override
+    public void updateTask(String updateField, String updateInfo) throws DukeException {
+        if (updateField.equals("/des")) {
+            this.description = updateInfo;
+        } else if (updateField.equals("/by")) {
+            String timeInfo = updateInfo;
+            timeInfo = changeWordToDate(timeInfo);
+            if (!checkTimeForm(timeInfo)) {
+                throw new TimeFormatException();
+            }
+            this.by = LocalDate.parse(timeInfo);
+        } else {
+            throw new WrongUsageException("use /by to identify update.");
+        }
+    }
+
+    private String changeWordToDate(String time) {
```

Comment:
This method seems like a duplicate. I wonder what is the rationale of having it duplicated?